common/lib/dependabot/update_checkers/version_filters.rb

```diff
@@ -0,0 +1,17 @@
+module Dependabot
+  module UpdateCheckers
+    module VersionFilters
+      def filter_vulnerable_versions(versions_array)
+        versions_array.reject do |v|
+          security_advisories.any? do |a|
+            if v.is_a?(Gem::Version)
```

Comment:
I wonder if we should transform the argument passed in here to always be a Gem::Version in the package managers that don't expose this?

Reply:
The tricky thing is we'd then also _return_ a `Gem::Version`, but in some cases we're filtering a set of "version details" hashes. We _could_ then filter those using the array `Gem::Version` returned, but that feels a little off?

Reply:
Aye, seems find as is 👍